### PR TITLE
Fix bug regarding serializing custom exceptions which occurs in multiprocessing context

### PR DESCRIPTION
**Pull request by basnijholt**
### Summary

The bug can be reproduced with:
```python
from qiskit_ionq.exceptions import IonQAPIError
import pickle
err = IonQAPIError(message='hi', status_code=400, headers='hi', body='hi', error_type='hi')
pickle.loads(pickle.dumps(err))
```
which gives:
```python
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[5], line 4
      2 import pickle
      3 err = IonQAPIError(message='hi', status_code=400, headers='hi', body='hi', error_type='hi')
----> 4 pickle.loads(pickle.dumps(err))

TypeError: IonQAPIError.__init__() missing 4 required positional arguments: 'status_code', 'headers', 'body', and 'error_type'
```

### CODE REVIEW

**Comment by amritpoudelionq** [comment]
@basnijholt also specify OS?

**Review by Cynocracy — APPROVED** [review]

**Comment by basnijholt** [comment]
@amritpoudelionq, the problem should occur on any OS.

File: qiskit_ionq/exceptions.py [hunk 1]
```diff
@@ -168,18 +168,12 @@ def __str__(self):
             f"error_type={self.error_type!r})"
         )
 
-    def __getstate__(self) -> object:
+    def __reduce__(self):
```
Comment (amritpoudelionq) [inline qiskit_ionq/exceptions.py:171]: do we understand why __reduce__ works? __get__ and __set__ are pretty standard way for serialization?

Comment (splch) [reply qiskit_ionq/exceptions.py:171]: i just tested this out and it seems like `__getstate__` and `__setstate__` are relying on an instantiated object (which is why we see the error `TypeError: IonQAPIError.__init__() missing 4 required positional arguments: 'status_code', 'headers', 'body', and 'error_type'`) but `__reduce__` returns the class and arguments to instantiate it.

**Pull request merged**